packages/react-components/src/components/utils/testUtils.tsx

```diff
@@ -52,6 +52,7 @@ export const renderWithLocalization = (
 export const createTestLocale = (testStrings: PartialDeep<ComponentStrings>): ComponentLocale => {
   const strings: ComponentStrings = COMPONENT_LOCALE_EN_US.strings;
   Object.keys(testStrings).forEach((key: string) => {
+    // @ts-expect-error TODO: fix noImplicitAny error here

```

Comment:
This is another important one that I believe is hiding an error, looking at the component strings we have a component string that can't be spread:
![image](https://github.com/Azure/communication-ui-library/assets/2684369/e73e3afb-5f53-4c21-937d-752bbe36a009)

Reply:
meaning `...testStrings[key]` or `...strings[key]` should throw for this value

Reply:
Or perhaps not throw.. will spread out the string, e.g. if `AttendeeRole` was `Consumer`, after this merge the component strings would look like:
```
{
  ...,
  AttendeeRole: {
    0: 'C',
    1: 'o',
    2: 'n',
    etc....
}
```

Reply:
This one was a real tough one that I had lots of difficulty trying to noImplicitAny. 
Even using "as" (which I know we should now be avoiding) I could not make it work. If you have time, would love to work together on this one as I believe there is similar code else where in our repo.

Reply:
This one is tough. We have a an object, that depending on the _key_ has a different value _type_.
Typescript can't handle this well:

```ts
interface myObjectType {
  'james': number;
  'edward': string;
}

/// given a key determined at runtime (e.g. in a Object.keys()) lets get the value:
const key = randomNumber === 1 ? 'james' : 'edward';
const a = myObject[key]; // a is correctly 'number' | 'string' because key could be 'james' or 'edward' at runtime

/// now lets just reassign that value right back to the object - should be simple right?
myObject[key] = value;
/// typescript will complain!

/// And typescript is right to.
/// The value could either be a number or a string, i.e. has type `number | string`.
/// There is no compilation step to determine that the type is string when the key is Edward, the type is _always_ `number | string`
/// Thus `myObject[key]` cannot be assigned to  `number | string`, it has to be `number` OR `string`, not the union of the two. 
/// i.e., it can't allow this assignment as it can't know that the value type correctly correlates with the runtime key value
```

So all that is to say, this is a case where `as` is needed and we basically have some "unsafe" code, sth like:
```ts
function getKeys<T extends object>(obj: T): (keyof T)[] {
  return Object.keys(obj) as Array<keyof T>;
}

/**
 * @private
 */
export const createTestLocale = (testStrings: ComponentStrings): ComponentLocale => {
  const strings = COMPONENT_LOCALE_EN_US.strings;
  // deep merge
  getKeys(testStrings).forEach((key) => {
    // mark the value as unknown because the type changes based on the key.
    // this is unsafe at runtime as we could assign the wrong type based on the key here.
    // but typescript isn't smart enough to know that the key used across each access will result in the same type
    (strings as Record<keyof ComponentStrings, unknown>)[key] = { ...strings[key], ...testStrings[key] };
  });
  return { strings };
};
```